File: NextcloudTalk/PlaceholderView.m
/**
 * SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

#import "PlaceholderView.h"

#import "NCAppBranding.h"

@interface PlaceholderView ()

@property (strong, nonatomic) IBOutlet UIView *contentView;

@end

@implementation PlaceholderView

- (instancetype)init
{
    self = [super init];
    
    if (self) {
        [[NSBundle mainBundle] loadNibNamed:@"PlaceholderView" owner:self options:nil];
        
        [self addSubview:self.contentView];
        
        self.contentView.frame = self.bounds;
    }
    
    return self;
}

- (instancetype)initForTableViewStyle:(UITableViewStyle)style
{
    self = [self init];
    
    if (self && (style == UITableViewStyleGrouped || style == UITableViewStyleInsetGrouped )) {
        self.contentView.backgroundColor = [UIColor systemGroupedBackgroundColor];
        self.placeholderView.backgroundColor = [UIColor systemGroupedBackgroundColor];
    }
    
    return self;
}

- (void)setImage:(UIImage *)image
{
    UIImage *placeholderImage = [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
    [self.placeholderImage setImage:placeholderImage];
    [self.placeholderImage setContentMode:UIViewContentModeScaleAspectFit];
    [self.placeholderImage setTintColor:[NCAppBranding placeholderColor]];
}

@end

File: NextcloudTalk/User Interface/PlaceholderView.swift
//
// SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
// SPDX-License-Identifier: GPL-3.0-or-later
//

import UIKit

@objcMembers class PlaceholderView: UIView {

    @IBOutlet var contentView: UIView!
    @IBOutlet weak var placeholderView: UIView!
    @IBOutlet weak var placeholderImage: UIImageView!
    @IBOutlet weak var placeholderTextView: UITextView!
    @IBOutlet weak var loadingView: UIActivityIndicatorView!

    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        commonInit()
    }

    // The style is unused, kept because every call site passes one
    convenience init(for style: UITableView.Style) {
        self.init(frame: .zero)
    }

    private func commonInit() {
        Bundle.main.loadNibNamed("PlaceholderView", owner: self, options: nil)

        self.addSubview(self.contentView)

        self.contentView.frame = self.bounds
    }

    func setImage(_ image: UIImage?) {
        self.placeholderImage.image = image?.withRenderingMode(.alwaysTemplate)
        self.placeholderImage.contentMode = .scaleAspectFit
        self.placeholderImage.tintColor = NCAppBranding.placeholderColor()
    }
}
